src/Gui/SkeletonBasedAnimation/SkeletonBasedAnimationUI.cpp

```diff
@@ -0,0 +1,475 @@
+    return 6;
+}
+
+QAction* SkeletonBasedAnimationUI::getAction( int i ) {
+    switch ( i )
+    {
+    case 0:
+        return ui->actionXray;
+    case 1:
+        return ui->actionLBS;
+    case 2:
+        return ui->actionDQS;
+    case 3:
+        return ui->actionCoR;
+    case 4:
+        return ui->actionSTBSLBS;
+    case 5:
+        return ui->actionSTBSDQS;
+    }
+    return nullptr;
+}
+
+void SkeletonBasedAnimationUI::postLoadFile( Engine::Scene::Entity* entity ) {
+    // reset current data and ui
+    m_selection       = Engine::Scene::ItemEntry();
+    m_currentSkeleton = nullptr;
+    m_currentSkinnings.clear();
+    ui->tabWidget->setEnabled( false );
+    ui->m_skinning->setEnabled( false );
+    // register the animation keyframes of the first animation into the timeline
+    auto c = std::find_if(
+        entity->getComponents().begin(), entity->getComponents().end(), []( const auto& cmpt ) {
+            return ( cmpt->getName().compare( 0, 3, "AC_" ) == 0 );
+        } );
+    if ( m_timeline )
+    {
+        if ( c != entity->getComponents().end() )
+        {
+            auto skel  = static_cast<Ra::Engine::Scene::SkeletonComponent*>( ( *c ).get() );
+            auto& anim = skel->getAnimation( skel->getAnimationId() );
+            const auto& boneMap = skel->getBoneRO2idx();
+            for ( size_t j = 0; j < anim.size(); ++j )
+            {
+                auto it = std::find_if( boneMap->begin(), boneMap->end(), [j]( const auto& b ) {
+                    return b.second == j;
+                } );
+                if ( it == boneMap->end() ) { continue; } // end bone
+
+                m_timeline->registerKeyFramedValue(
+                    it->first,
+                    KeyFramedValueController(
+                        &anim[j],
+                        "Animation_" + skel->getSkeleton()->getLabel( uint( j ) ),
+                        [&anim, j, skel]( const Scalar& t ) {
+                            anim[j].insertKeyFrame(
+                                t,
+                                skel->getSkeleton()->getPose( HandleArray::SpaceType::LOCAL )[j] );
+                        } ) ); // no update callback here
+            }
+        }
+        // update the timeline display interval to the bounding interval of all anims
+        Scalar startTime = std::numeric_limits<Scalar>::max();
+        Scalar endTime   = 0;
+        for ( const auto& animComp : entity->getComponents() )
+        {
+            if ( animComp->getName().compare( 0, 3, "AC_" ) == 0 )
+            {
+                auto skel   = static_cast<Ra::Engine::Scene::SkeletonComponent*>( animComp.get() );
+                auto [s, e] = skel->getAnimationTimeInterval();
+                startTime   = std::min( startTime, s );
+                endTime     = std::max( endTime, e );
+            }
+        }
+        m_timeline->onChangeStart( startTime );
+        m_timeline->onChangeEnd( endTime );
```

Comment:
Isn't this redundant with what is done in `SkeletonBasedAnimationSystem::handleAssetLoading`? (I would use only the one here)

Reply:
This is not redundant. In `SkeletonBasedAnimationSystem::handleAssetLoading` the engine is configured. here, the timeline is updated. As the engine as no information about timeline, this must be done like this.